File: 2073-time-needed-to-buy-tickets/2073-time-needed-to-buy-tickets.java
class Solution {
    public int timeRequiredToBuy(int[] tickets, int k) {
        int time = 0;
        while (tickets[k] > 0) {
            for (int i = 0; i < tickets.length; i++) {
                if (tickets[k] == 0) break;
                if (tickets[i] > 0) {
                    tickets[i]--;
                    time++;
                }
            }
        }
        return time;
    }
    
}

File: 2073-time-needed-to-buy-tickets/2073-time-needed-to-buy-tickets.kt
class Solution {
    fun timeRequiredToBuy(t: IntArray, k: Int): Int {
        val n = t.size
        var result = 0
        for (i in 0..n-1) {
             result += minOf(t[i], if (i > k) t[k] - 1 else t[k])
        }
        return result
    }
}